ecosystem/typescript/sdk/src/plugins/aptos_token.ts

```diff
@@ -0,0 +1,466 @@
+      creator,
+      "set_name",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), name],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Set token URI
+   * @param creator Creator account
+   * @param token Token address
+   * @param uri Token uri
+   * @returns The hash of the transaction submitted to the API
+   */
+  async setTokenURI(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    uri: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "set_uri",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), uri],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Add token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key for storing on-chain property
+   * @param type the type of property value
+   * @param value the property value to be stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async addTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    type: string,
+    value: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "add_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key, type, getSinglePropertyValueRaw(value, type)],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Remove token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async removeTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "remove_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Update token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key stored on-chain
+   * @param type the property typed stored on-chain
+   * @param value the property value to be stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async updateTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    type: string,
+    value: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "update_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key, type, getSinglePropertyValueRaw(value, type)],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Transfer a token ownership
```

Comment:
Let's add some documentation to note that this would only work if the token is not frozen (i.e. owner transfer is not disabled such as for soul bound tokens)